graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/AtlasJanusElement.java

```diff
@@ -130,19 +157,33 @@ public void removePropertyValue(String propertyName, Object propertyValue) {
     @Override
     public void removeAllPropertyValue(String propertyName, Object propertyValue) {
         Iterator<? extends Property<Object>> it = getWrappedElement().properties(propertyName);
-
+        RequestContext context = RequestContext.get();
         while (it.hasNext()) {
             Property currentProperty      = it.next();
             Object   currentPropertyValue = currentProperty.value();
 
             if (Objects.equals(currentPropertyValue, propertyValue)) {
                 currentProperty.remove();
+                //fill the map for the entityGuid and remove value
+                if (propertyName.startsWith(INTERNAL_PROPERTY_KEY_PREFIX)) {
+                    String entityGuid = this.getProperty(GUID_PROPERTY_KEY, String.class);
+                    if (StringUtils.isNotEmpty(entityGuid)) {
+                        if (context.getAllInternalAttributesMap().get(entityGuid) != null) {
+                            context.getAllInternalAttributesMap().get(entityGuid).put(propertyName, null);
+                        } else {
+                            Map<String, Object> map = new HashMap<>();
+                            map.put(propertyName, null);
+                            context.getAllInternalAttributesMap().put(entityGuid, map);
+                        }
+                    }
+                }
```

Comment:
From method name I think intention is to remove all items from value list

E.g
Imagine __meaningNames current has values ["term_0", "term_1"],
Someone updates asset to remove both term_0, term_1

I will check more on this method & get back